File: Javascript/Array/M-419. Battleships in a Board.ts
function countBattleships(board: string[][]): number {
  let battleships = 0;

  for (let i = 0; i < board.length; i++) {
    for (let j = 0; j < board[i].length; j++) {
      if (i > 0 && board[i - 1][j] === 'X') continue;
      if (j > 0 && board[i][j - 1] === 'X') continue;
      if (board[i][j] === 'X') battleships++;
    }
  }

  return battleships;
}

File: Python/Array/M-419. Battleships in a Board.py
class Solution:
    def countBattleships(self, board: List[List[str]]) -> int:
        battleships = 0
        n, m = len(board), len(board[0])
        for i in range(n):
            for j in range(m):
                if i > 0 and board[i - 1][j] == "X":
                    continue
                if j > 0 and board[i][j - 1] == "X":
                    continue
                if board[i][j] == "X":
                    battleships += 1

        return battleships
